src/java.base/share/classes/sun/security/util/Hybrid.java

```diff
@@ -0,0 +1,452 @@
+import java.util.Arrays;
+import java.util.Locale;
+
+public class Hybrid {
+
+    public record SecretKeyImpl(SecretKey k1, SecretKey k2) implements SecretKey {
+        @Override
+        public String getAlgorithm() {
+            return "Hybrid";
+        }
+
+        @Override
+        public String getFormat() {
+            return null;
+        }
+
+        @Override
+        public byte[] getEncoded() {
+            return null;
+        }
+    }
+
+    private static AlgorithmParameterSpec getSpec(String name) {
+        if (APS.isGenericEC(name)) {
+            return new ECGenParameterSpec(name);
+        } else {
+            return new NamedParameterSpec(name);
+        }
+    }
+
+    private static KeyPairGenerator getKeyPairGenerator(String name) throws
+            NoSuchAlgorithmException {
+        if (APS.isGenericEC(name)) {
+            name = "EC";
+        }
+        return KeyPairGenerator.getInstance(name);
+    }
+
+    private static KeyFactory getKeyFactory(String name) throws
+            NoSuchAlgorithmException {
+        if (APS.isGenericEC(name)) {
+            name = "EC";
+        }
+        return KeyFactory.getInstance(name);
+    }
+
+    /**
+     * Returns a KEM instance for each side of the hybrid algorithm.
+     * For traditional key exchange algorithms, we use the DH-based KEM
+     * implementation provided by DH.PROVIDER.
+     * For ML-KEM post-quantum algorithms, we obtain a KEM instance
+     * using the given algorithm name.
+     */
+    private static KEM getKEM(String name) throws NoSuchAlgorithmException {
+        if (APS.isGenericEC(name) || APS.isXDH(name)) {
+            return KEM.getInstance("DH", DH.PROVIDER);
+        } else {
+            return KEM.getInstance(name);
+        }
+    }
+
+    public static class KeyPairGeneratorImpl extends KeyPairGeneratorSpi {
+        private final KeyPairGenerator left;
+        private final KeyPairGenerator right;
+        private final AlgorithmParameterSpec leftSpec;
+        private final AlgorithmParameterSpec rightSpec;
+
+        public KeyPairGeneratorImpl(String leftAlg, String rightAlg)
+                throws NoSuchAlgorithmException  {
+            left = getKeyPairGenerator(leftAlg);
+            right = getKeyPairGenerator(rightAlg);
+            leftSpec = getSpec(leftAlg);
+            rightSpec = getSpec(rightAlg);
+
+            try {
+                left.initialize(leftSpec);
+                right.initialize(rightSpec);
+            } catch (InvalidAlgorithmParameterException iape) {
+                throw new ProviderException("Invalid algorithm parameters " +
+                        "for hybrid keypair generator", iape);
+            } catch (Exception e) {
+                throw new ProviderException("Failed to initialize hybrid " +
+                        "keypair generator", e);
+            }
+        }
+
+        @Override
+        public void initialize(int keysize, SecureRandom random) {
```

Comment:
Is this method actually called?  (I don't see it in the `KEMKeyExchange`)  

I know the method is needed to fully extend the `KeyPairGeneratorSpi`, but it looks like we're doing the actual initialization of both sides in the constructor. 

This duplicate code could probably be removed.